Formula/dhall.rb

```diff
@@ -0,0 +1,26 @@
+require "language/haskell"
+
+class Dhall < Formula
+  include Language::Haskell::Cabal
+
+  desc "Compiler for the Dhall language"
+  homepage "https://github.com/dhall-lang/dhall-haskell"
+  url "https://hackage.haskell.org/package/dhall-1.18.0/dhall-1.18.0.tar.gz"
+  sha256 "b528ecff4f757fa085f61636ffb8a6eb90c761106d7d20bd735bee48a6abab94"
+  head "https://github.com/dhall-lang/dhall-haskell.git"
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+
+  def install
+    if build.head?
+      install_cabal_package "dhall"
+    else
+      install_cabal_package
+    end
+  end
+
+  test do
+    assert_match "{=}", pipe_output("#{bin}/dhall format", "{=}", 0)
```

Comment:
Are we only checking that it outputs back what it is given in input? Can we have a somewhat stronger test?

Reply:
Added some more tests; unsure how much really needs to be tested here